java/core/src/java/org/apache/orc/impl/ParserUtils.java

```diff
@@ -281,7 +282,7 @@ public void visit(TypeDescription type, int posn) {
   public static TypeDescription findSubtype(TypeDescription schema,
                                             ParserUtils.StringPosition source,
                                             boolean isSchemaEvolutionCaseAware) {
-    TypeFinder result = new TypeFinder(removeAcid(schema));
+    TypeFinder result = new TypeFinder(schema);
```

Comment:
I am uncomfortable changing this here. This method is being used in multiple places and we need to ensure in all the places the source includes the prefix of `row`.

These are the following two broad approaches I see here.

1. Ignore the`row` `StructColumnVector` in all processing, which seems to be case with the current code when removing acid schema.
2. Consider the `row` column vector in all of the processing (partially implemented)

I am leaning towards the first option so as not to change the behavior of the APIs. e.g. if previously one was calling with col1, now they are expected to call as row.col1 which is a change in behavior.

This is also feels like it will minimize the amount of code that needs to be changed.

Reply:
Let me know what you think.

Reply:
I tried with the following to get the test to pass.

```diff
diff --git a/java/core/src/java/org/apache/orc/impl/ParserUtils.java b/java/core/src/java/org/apache/orc/impl/ParserUtils.java
index 6491fb8ba..b56fa35b3 100644
--- a/java/core/src/java/org/apache/orc/impl/ParserUtils.java
+++ b/java/core/src/java/org/apache/orc/impl/ParserUtils.java
@@ -411,18 +411,22 @@ public class ParserUtils {
     private final ColumnVector[] result;
     private int resultIdx = 0;

-    ColumnFinder(TypeDescription schema, VectorizedRowBatch batch, int levels) {
+    ColumnFinder(TypeDescription schema, ColumnVector[] cols, int levels) {
       if (schema.getCategory() == TypeDescription.Category.STRUCT) {
-        top = batch.cols;
+        top = cols;
         result = new ColumnVector[levels];
       } else {
         result = new ColumnVector[levels + 1];
-        current = batch.cols[0];
+        current = cols[0];
         top = null;
         addResult(current);
       }
     }

+    ColumnFinder(TypeDescription schema, VectorizedRowBatch batch, int levels) {
+      this(schema, batch.cols, levels);
+    }
+
     private void addResult(ColumnVector vector) {
       result[resultIdx] = vector;
       resultIdx += 1;
@@ -459,7 +463,10 @@ public class ParserUtils {
                                                  boolean isCaseSensitive,
                                                  VectorizedRowBatch batch) {
     List<String> names = ParserUtils.splitName(source);
-    ColumnFinder result = new ColumnFinder(schema, batch, names.size());
+    TypeDescription schemaToUse = removeAcid(schema);
+    ColumnVector[] cols = SchemaEvolution.checkAcidSchema(schema)
+      ? ((StructColumnVector) batch.cols[batch.cols.length - 1]).fields : batch.cols;
+    ColumnFinder result = new ColumnFinder(schemaToUse, cols, names.size());
     findColumn(removeAcid(schema), names, isCaseSensitive, result);
     return result.result;
   }
```

Reply:
@pavibhai thanks for the idea! I agree with you, changed accordingly.